adafruit_esp32spi/adafruit_esp32spi_socket.py

```diff
@@ -52,14 +52,23 @@ class socket:
 
     # pylint: disable=too-many-arguments
     def __init__(
-        self, family=AF_INET, type=SOCK_STREAM, proto=0, fileno=None, socknum=None
+        self,
+        family=AF_INET,
+        type=SOCK_STREAM,
+        proto=0,
+        fileno=None,
+        socknum=None,
+        remote_ip=None,
+        remote_port=None,
```

Comment:
```suggestion
```

Reply:
Oh that makes sense @tannewt!  I will work on these changes as soon as I get a second and will test and push an update.  Thank you again.

Reply:
No rush and no problem! Thanks!

Reply:
@tannewt I have implemented the changes.  I wanted ask specifically how you would like this tested.  Here is the result in the  REPL.
`>>> socket.socket.
close           connect         connected       read
readline        recv            send            settimeout
write           socknum         available       getpeername`

I obviously cant call `socket.socket.getpeername()` so if you have a quick test I can code up I will begin testing so that I know what you are looking for specifically.  Any simple example will do thanks!

Reply:
I'd imagine you need to create a socket and connect before calling it. The new requests library can work with CPython sockets so that's probably the best place to start.

Reply:
@tannewt I created an access point called `cool` and then I connected my iPhone to it.  It successfully connected however when I call `socket.socket.getpeername()` i get `TypeError: function takes 1 positional arguments but 0 were given` as it should be returning the peer ip of `192.168.4.2` as that is the iPhone and the server is `192.168.4.2`. 

Any suggestions on how to better check the `getpeername` call?

Reply:
You actually need to connect a TCP socket and then call .getpeername() on it. The error is because you are calling it on the class, not an instance, and therefore no `self` is passed.

Why are you adding this overall? It doesn't seem like you have a use for it.

Reply:
Great question @tannewt.  Over a year ago when I did the original PR I had issues where I had the device set up as an Access Point and if I accidently logged in from multiple browsers on various machines it would crash. 

This original PR was designed to check for that remote IP and if it was not 192.168.4.2 then don't connect.  It was to add the ability to get this information if a developer wanted to implement functionality such as this.

I am ok with closing the PR however.  Thoughts?

Reply:
This is what I tried to do originally as well.  I can't seem to even create a socket object here.
`>>> import adafruit_esp32spi.adafruit_esp32spi_socket as socket
>>> s = socket.socket()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_socket.py", line 62, in __init__
AttributeError: 'NoneType' object has no attribute 'get_socket'
>>> `

Reply:
With the current ESP32-SPI implementation you need to call `set_interface` on the socket module: https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi_socket.py#L24 I believe you pass it the ESP32SPI object.

I could see this PR being helpful but if no one is using it enough to test, then we could just close for now.